src/main/java/net/neoforged/neoforge/event/village/WandererTradesEvent.java

```diff
@@ -21,10 +22,20 @@
 public class WandererTradesEvent extends Event {
     protected List<ItemListing> generic;
     protected List<ItemListing> rare;
+    private final RegistryAccess registryAccess;
 
+    /**
+     * @deprecated Use {@link #WandererTradesEvent(List, List, RegistryAccess)} instead
+     */
+    @Deprecated(forRemoval = true, since = "1.21")
     public WandererTradesEvent(List<ItemListing> generic, List<ItemListing> rare) {
+        this(generic, rare, RegistryAccess.EMPTY);
+    }
+
+    public WandererTradesEvent(List<ItemListing> generic, List<ItemListing> rare, RegistryAccess registryAccess) {
```

Comment:
Should we perhaps mark this with `@ApiStatus.Internal`? I can't think of a reason why a mod would construct this event, since NeoForge does this already.